zerver/views/streams.py

```diff
@@ -125,6 +127,15 @@ def check_if_removing_someone_else(user_profile: UserProfile,
     else:
         return principals[0] != user_profile.email
 
+def parse_message_retention_days(value: Union[int, str]) -> Optional[int]:
+    if value == "forever":
+        return -1
+    if value == "realm_default":
+        return None
+    if isinstance(value, str) or value < 0:
+        raise JsonableError(_("Invalid value of message_retention_days"))
```

Comment:
We should use the standard exception from `zerver/lib/request.py`; I'll push back with this change (you'll need to fix the tests).